### PR TITLE
add vpc filter to oidc listdatabases request

### PR DESCRIPTION
**Pull request by GavinFrazar**
Part of:
- https://github.com/gravitational/teleport/issues/41004

Majority of the diff is for protobuf gen.

This PR adds an optional VPC ID filter to our AWS OIDC listdatabases endpoint.
This will enable the web UI AWS database discovery integration to filter the databases by VPC ID as we transition the integration to configure db discovery by region/vpc instead of just the region.

Since AWS doesn't support a VPC ID filter in its API, I came up with a scheme where this endpoint will now try to return at least `(page size)` items, up to a maximum of `2*(page size) -1` items, and changed the requested page size we use to 50 (half of the default) to keep the response size no larger than it was before.

This is to handle the (unlikely) scenario where a customer's AWS account has hundreds of databases in one region spread across multiple VPCs, in a simplistic way that doesn't require caching to handle page size misalignment issues.
Without this scheme, the caller (our web ui code) would need to handle cases where fewer than page size items are returned with a non-empty next page token - the caller would just call the endpoint repeatedly in that case too.

Changes to actually pass a VPC ID selected from the web ui will be in a future PR.

### CODE REVIEW

**Comment by GavinFrazar** [comment]
@greedy52 I'd like to backport these changes to v16 at least, what do you think?

**Review by marcoandredinis — APPROVED** [review]

File: lib/integrations/awsoidc/listdatabases.go [hunk 1]
```diff
@@ -104,6 +106,12 @@ func NewListDatabasesClient(ctx context.Context, req *AWSClientRequest) (ListDat
 	return newRDSClient(ctx, req)
 }
 
+// pageSize is half the default RDS list input page size (100).
+// We filter by VPC membership after the API call and try to return pageSize
+// items but can return up to pageSize*2 -1 items, so we use a smaller page size
+// than the default.
+var pageSize int32 = 50
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply lib/integrations/awsoidc/listdatabases.go]: makes sense, renamed in: https://github.com/gravitational/teleport/compare/f96e4e05bec6c2102ad77029b2d8c7a7b80f3601..e33ee68071f3bf9d848870b8f07bc24e0cfa11db

**Review by greedy52 — APPROVED** [review]
LGTM as long as web UI side can handle the odd numbers. Like first time it may return 55 and second time 24.

The RDS default quota is 40 cluster/instance per region, so maybe we are overthinking this =D.

**Comment by GavinFrazar** [comment]
force pushed to regenerate conflicted protobuf gen.

**Review by avatus — APPROVED** [review]

**Pull request merged**